### DISCUSSION TITLE
Single domain vs multiple domain for installation

### DISCUSSION THREAD

**Discussion by gramakri** [minio/minio#12780] [Q&A] [answered]
I have been looking into updating the Cloudron minio package to a recent version. Recent minio has integrated the new console and there are two ports: the console UI port and the API port. Is there any recommendation between these three deployment approaches:

* Have a reverse proxy (single domain) in the front that proxies to these two ports. My understanding is that if I do this some bucket names like "static" won't work anymore because that path is taken by the console UI. Alternately, maybe there is a way to serve up the console in a subpath like say `/console` (and now this bucket name is reserved, but that's OK ) ? 

* Expose them in different ports (single domain). This poses a challenge that it seems that `--certs-dir` switches both the ports to be HTTPS. Is there way to say just the API port has to use the certs? (because the console UI already has a reverse proxy in front of it).

* Use completely separate domains for the console and API.

Is there any preferred deployment between the above choices?

Thanks!

**Comment by northway** [comment, marked as answer]
After some testing I implemented the `Use completely separate domains for the console and API` way with the `--console-address ":9001"` option.

You can separate them in the background, port 9000 and 9001 are masked with two nginx proxy vhost.

The clients could use standard HTTPS port because I'm using a proxy and it's easier to explain to the customers what's the difference between the two URL.

**Reply by gramakri** [reply]
Yeah, your approach seems to be the completely future proof way. It's entirely possible that the console code takes up more subpaths in the future and it conflicts with existing bucket names.
